p2p/src/peer_manager/mod.rs

```diff
@@ -690,13 +812,80 @@ where
         Ok(())
     }
 
+    /// Get the number of active peers
+    pub fn active_peer_count(&self) -> usize {
+        self.peers.len()
+    }
+
+    /// Returns short info about all connected peers
+    pub fn get_connected_peers(&self) -> Vec<ConnectedPeer> {
+        self.peers.values().map(Into::into).collect()
+    }
+
+    /// Selects requested count of connected peer ids randomly.
+    ///
+    /// It can be used to distribute data in the gossip protocol
+    /// (for example, to relay announced addresses to a small group of peers).
+    pub fn random_peer_ids(&self, count: usize) -> Vec<T::PeerId> {
+        // There are normally not many connected peers, so iterating over the whole list should be OK
+        let all_peer_ids = self.peers.keys().cloned().collect::<Vec<_>>();
```

Comment:
Please throw in a TODO here to optimize this. Always do this when we have to unnecessarily clone instead of pick random items. And btw, I really think implementing this is not that hard, but I'm not gonna pressure for it now.

Reply:
Sure, done.

Reply:
Optimize. We use American English, not British :-)